packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -202,23 +214,25 @@ class ResidentWebRunner extends ResidentRunner {
     // Cleanup old subscriptions. These will throw if there isn't anything
     // listening, which is fine because that is what we want to ensure.
     try {
-      await _debugConnection?.vmService?.streamCancel('Stdout');
+      await _vmService?.streamCancel('Stdout');
```

Comment:
What's the difference between `_vmService` being `null` and `supportsServiceProtocol` being `false`? I'm not exactly sure why we null-check here but not inside `if (supportsServiceProtocol)`.

Reply:
I guess they're actually the same, I could collapse them into one check